packages/core/src/exchanges/ssr.ts

```diff
@@ -109,7 +109,6 @@ const serializeResult = (
   includeExtensions: boolean
 ): SerializedResult => {
   const serialized: SerializedResult = {
-    data: JSON.stringify(result.data),
```

Comment:
I'd prefer removing the `if` branch then as `JSON.stringify(undefined)` still has a result.

Reply:
On the other hand, calling `JSON.stringify` with undefined is slower
https://jsperf.app/vuwofo

Reply:
Yes, but the property missing isn't really according to spec so we might even want to default that to null https://graphql.github.io/graphql-over-http/draft/#sec-application-json

Reply:
The `SerializedResult` type allows the data to be a `string` only or `undefined`. `deserializeResult` will return `undefined` if there is no data.
https://github.com/urql-graphql/urql/blob/6406fd5fa30180648439e9f9ba9f18caf16e2497/packages/core/src/exchanges/ssr.ts#L154

Looking at the implementation closer, I'd like to use something more efficient than `JSON.stringify`, like [devalue](https://github.com/Rich-Harris/devalue). Because I can reduce 2mb JSON to 1.1MB, I just checked this out